redirect.php

```diff
@@ -81,8 +82,14 @@ function handle_redirect() {
 		return;
 	}
 
+	// If current domain and domain mapping are the same, exit early.
+	$domain = $mapping->get_site()->domain;
+	if ( $domain == $_SERVER['HTTP_HOST'] ) {
+		return;
+	}
+
 	// Use blogs table domain as the primary domain
-	wp_redirect( 'http://' . $mapping->get_site()->domain . esc_url_raw( $_SERVER['REQUEST_URI'] ), 301 );
+	wp_redirect( set_url_scheme( "http://" . $domain . esc_url_raw( $_SERVER['REQUEST_URI'] ) ), apply_filters( 'mercator.redirect.status.code', 301 ) );
```

Comment:
Generally prefer to avoid long lines - can you move the `apply_filters()` call to a line above and assign it to a var. Can also document the filter then.

One other thing - should we validate the redirect code? Will it always be `301` or `302`? If it's binary why not have the filter return a simple `true` or `false` and call it `mercator.redirect.permanent`. Can be used like:

```php
add_filter( 'mercator.redirect.permanent', '__return_false' );
```